### DISCUSSION TITLE
Feature request: Support Loki log volume histogram in custom dashboards

### DISCUSSION THREAD

**Discussion by dvidben** [grafana/grafana#46600] [Dashboards, Panels & Visualizations]
In the Loki explore view, grafana offers a histogram to show the log volume in a histogram supporting following features:

- log aggregations over time
- split on log level. 
- log levels color palette described here: https://grafana.com/docs/grafana/latest/explore/logs-integration/ (section titled "Log level").

However, current available charts in grafana do not offer the same capabilities to create a dashboard with the same features:
1. Bucketing based on time is limited in other charts (i.e: histogram, heatmap)
2. Log-level based color palette is not available (time series using bars - color palette is available for values but not legends, histogram, heatmap)

This request is to support adding a log-volume histogram with the same capabilities in custom dashboards (either by adding a specific new control or extending the current capabilities in the timeseries or histogram charts)

**Comment by jonathanrgreene** [comment]
I'm actually surprised this doesn't work out of the gate. Am I missing something? Is there some way to make this work?

**Reply by mafcentauri** [reply]
Me also, literally what I want is shown to me by Loki but it's not possible to simply add that to a dashboard

**Comment by ndlanier** [comment]
I would really love it if this was implemented!

**Comment by svennergr** [comment]
You can create a panel like that with the following settings:
https://play.grafana.org/d/f03dfa36-f578-4f4e-ac5e-2bce58476674/log-volume?orgId=1&from=1683531030209&to=1683534630209

Panel JSON:

```json
{
  "datasource": {
    "type": "loki",
    "uid": "zXvfRSSVz"
  },
  "fieldConfig": {
    "defaults": {
      "custom": {
        "drawStyle": "bars",
        "lineInterpolation": "stepBefore",
        "barAlignment": 1,
        "lineWidth": 1,
        "fillOpacity": 100,
        "gradientMode": "none",
        "spanNulls": false,
        "showPoints": "auto",
        "pointSize": 5,
        "stacking": {
          "mode": "normal",
          "group": "A"
        },
        "axisPlacement": "auto",
        "axisLabel": "",
        "axisColorMode": "text",
        "scaleDistribution": {
          "type": "linear"
        },
        "axisCenteredZero": false,
        "hideFrom": {
          "tooltip": false,
          "viz": false,
          "legend": false
        },
        "thresholdsStyle": {
          "mode": "off"
        },
        "lineStyle": {
          "fill": "solid"
        }
      },
      "color": {
        "mode": "palette-classic"
      },
      "mappings": [],
      "thresholds": {
        "mode": "absolute",
        "steps": [
          {
            "color": "green",
            "value": null
          }
        ]
      }
    },
    "overrides": []
  },
  "gridPos": {
    "h": 8,
    "w": 24,
    "x": 0,
    "y": 0
  },
  "id": 1,
  "interval": "1m",
  "options": {
    "tooltip": {
      "mode": "single",
      "sort": "none"
    },
    "legend": {
      "showLegend": true,
      "displayMode": "list",
      "placement": "bottom",
      "calcs": []
    }
  },
  "targets": [
    {
      "datasource": {
        "type": "loki",
        "uid": "zXvfRSSVz"
      },
      "editorMode": "code",
      "expr": "sum by (level) (count_over_time({container=\"tns-app\"} | logfmt [$__auto]))",
      "legendFormat": "{{level}}",
      "queryType": "range",
      "refId": "A"
    }
  ],
  "title": "Log Volume",
  "transformations": [],
  "type": "timeseries"
}
```

**Reply by MargaretKrutikova** [reply]
That is awesome, thank you! ❤️  That solves my use-case at least.

**Comment by v-zhuravlev** [comment]
The problem in the example above is that colors are not aligned with what you get in built in graph. You also can't sync logs panel contents by just clicking into graph legend (if there is a logs panel). 

Since this logs volume graph exists already, I wonder if it is easy enough to port the graph from 'explore' to the logs panel as an option (show/not show)?

**Comment by gelicia** [comment]
Hello! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

There is a good workaround for this issue, which is to use the panel code provided by @svennergr here https://github.com/grafana/grafana/discussions/46600#discussioncomment-5835693 

As for a more permanent solution, this discussion was converted into an issue, which is available here: https://github.com/grafana/grafana/issues/73430
